src/constraints/GeomMeanEpiCone.jl

```diff
@@ -5,146 +5,87 @@
+just for consistency with the GeomMeanHypoCone function.
 
-REFERENCE
-  Ported from CVXQUAD which is based on the paper: "Lieb's concavity
-  theorem, matrix geometric means and semidefinite optimization" by Hamza
-  Fawzi and James Saunderson (arXiv:1512.03401)
+## Reference
+
+Ported from CVXQUAD which is based on the paper: "Lieb's concavity theorem,
+matrix geometric means and semidefinite optimization" by Hamza Fawzi and James
+Saunderson (arXiv:1512.03401)
 """
-mutable struct GeomMeanEpiCone
-    A::AbstractExpr
-    B::AbstractExpr
+struct GeomMeanEpiConeSquare <: MOI.AbstractVectorSet
     t::Rational
-    size::Tuple{Int,Int}
+    side_dimension::Int
 
-    function GeomMeanEpiCone(
-        A::AbstractExpr,
-        B::AbstractExpr,
-        t::Rational,
-        fullhyp::Bool = true,
-    )
-        if size(A) != size(B)
-            throw(DimensionMismatch("A and B must be the same size"))
-        end
-        n = size(A)[1]
-        if size(A) != (n, n)
-            throw(DimensionMismatch("A and B must be square"))
-        end
-        if t < -1 || (t > 0 && t < 1) || t > 2
+    function GeomMeanEpiConeSquare(t::Rational, side_dimension::Int)
+        if !(-1 <= t <= 0 || 1 <= t <= 2)
             throw(DomainError(t, "t must be in the range [-1, 0] or [1, 2]"))
         end
-        return new(A, B, t, (n, n))
-    end
-
-    function GeomMeanEpiCone(
-        A::Value,
-        B::AbstractExpr,
-        t::Rational,
-        fullhyp::Bool = true,
-    )
-        return GeomMeanEpiCone(constant(A), B, t, fullhyp)
-    end
-
-    function GeomMeanEpiCone(
-        A::AbstractExpr,
-        B::Value,
-        t::Rational,
-        fullhyp::Bool = true,
-    )
-        return GeomMeanEpiCone(A, constant(B), t, fullhyp)
-    end
-
-    function GeomMeanEpiCone(
-        A::Value,
-        B::Value,
-        t::Rational,
-        fullhyp::Bool = true,
-    )
-        return GeomMeanEpiCone(constant(A), constant(B), t, fullhyp)
-    end
-
-    function GeomMeanEpiCone(
-        A::Union{AbstractExpr,Value},
-        B::Union{AbstractExpr,Value},
-        t::Integer,
-        fullhyp::Bool = true,
-    )
-        return GeomMeanEpiCone(A, B, t // 1, fullhyp)
+        return new(t, side_dimension)
     end
 end
 
-mutable struct GeomMeanEpiConeConstraint <: Constraint
-    T::AbstractExpr
-    cone::GeomMeanEpiCone
+MOI.dimension(set::GeomMeanEpiConeSquare) = 3 * set.side_dimension
 
-    function GeomMeanEpiConeConstraint(T::AbstractExpr, cone::GeomMeanEpiCone)
-        if size(T) != cone.size
-            throw(DimensionMismatch("T must be size $(cone.size)"))
-        end
-        return new(T, cone)
-    end
+head(io::IO, ::GeomMeanEpiConeSquare) = print(io, "∈(GeomMeanEpiConeSquare)")
 
-    function GeomMeanEpiConeConstraint(T::Value, cone::GeomMeanEpiCone)
-        return GeomMeanEpiConeConstraint(constant(T), cone)
+function Base.in(func::Tuple, set::GeomMeanEpiConeSquare)
```

Comment:
We should probably get rid of this as it will become type piracy once `GeomMeanEpiConeSquare` is moved to MOI

Reply:
maybe we can have syntax like `(A,B,C) in Constraint(MOI.GeomMeanEpiConeSquare())` instead, where `Constraint` would be owned by Convex

Reply:
Yes, this can be a follow up PR

Reply:
Let's just make this `GenericConstraint(::Tuple, ::GeometricMeanEpiConeSquare)` for now?

Reply:
Or `Base.in(f::AbstractExpr, set::MOI.AbstractSet) = GenericConstraint(f, set)`? Why do we need the tuple? Why not `vcat(T, A, B)`?

Reply:
It was just be able to throw the DimensionMismatch errors

Reply:
We can also just have some internal utility for now

Reply:
Like we did for `_add_vectorized_exp_cone`

Reply:
How about now? I've just gone for explicit `Convex.GenericConstraint`.